### PR TITLE
LLM scan

### PR DESCRIPTION
**Pull request by kevinmessiaen**
## Description

Added a new "business" detector that allow to scan generate functional issues for LLM tasks:

- Generate input dataset based on the function tasks
- Generate functional tests

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [x] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

### CODE REVIEW

File: giskard/core/suite.py [hunk 1]
```diff
@@ -492,7 +499,7 @@ def _remove_test_by_idx(self, idx: int):
 
     @remove_test.register
     def _remove_test_by_name(self, test_name: str):
```
Comment (Hartorn) [inline giskard/core/suite.py:501]: I guess mehod could be rename for clarity

File: giskard/llm/issues/__init__.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
+from typing import List, Dict
+
+from pydantic import BaseModel, Field
+
+from ..config import llm_config
+from ...scanner.llm.utils import LLMImportError
```
Comment (Hartorn) [inline giskard/llm/issues/__init__.py:6]: It may be more readable to use absolute import giskard.xx.

**Review by Hartorn — COMMENTED** [review]
Not much to say on code structure :)

**Comment by mattbit** [comment]
Detection part is in https://github.com/Giskard-AI/giskard/pull/1468

There are still commits in here that should be kept to align with the hub functionality.

**Comment by kevinmessiaen** [comment]
> Detection part is in #1468
> 
> There are still commits in here that should be kept to align with the hub functionality.

I'm looking to integrate them on the other branch

**Pull request closed without merging**